someSong/src/pages/home/home.ts

```diff
@@ -18,11 +19,23 @@ export class HomePage {
   userSubscription: any;
 
   constructor(public navCtrl: NavController,
-              private _user: User) {
-    this.user = { image: '' };
+              private _user: User,
+              private _answer: Answer) {
     this.userSubscription = this._user.currentUser.subscribe((data) =>
     {
       this.user = data;
+
+      var fullAnswers = new Array<any>();
+
+      for (let answer of this.user.answers)
```

Comment:
why not add a `getAllAnswers(userID)` and a `getAllQuestions(userID)` functions in the `answers` and `questions` providers?
And use it like this:
`this.user.answers = this._answer.getAllAnswers(this.user.userID) ;`